netutils/interface.py

```diff
@@ -1,8 +1,89 @@
 """Functions for working with interface."""
 
+import itertools
 from .variables import BASE_INTERFACES, REVERSE_MAPPING
 
 
+def interface_tail_addition(interface_number):
+    """Takes the tail end of the interface and sums it up to use for sorting a list of interfaces.
+
+    Args:
+        interface_number (str): Interface number I.E: 1/0/1, 1/0, 0/1
+
+    Returns:
+        int: Sum of number in interface tail with each one getting a priority of magnitude 10 from right to left.
+
+    Example:
+        >>> from netutils.interface import interface_tail_addition
+        >>> interface_tail_addition("1/0/6")
+        106
+        >>> interface_tail_addition("2/3/6")
+        236
+        >>>
+    """
+    priority_values = []
+    factor_value = 1
+    split_post_sub = []
+
+    split_pre_sub = split_interface(interface_number)[1].split("/")
+    for item in split_pre_sub:
+        if "." in item:
+            split_post_sub.append(item.split(".")[0])
+        else:
+            split_post_sub.append(item)
+
+    values = [int(i) for i in split_post_sub]
+    for i in reversed(values):
+        i = i * factor_value
+        factor_value = factor_value * 10
+        priority_values.insert(0, i)
+    return sum(priority_values)
+
+
+def list_alphabetical(interfaces, reverse=False):
+    """Takes list of interfaces and ensures they are in alphabetical order.
+
+    Args:
+        interfaces (list): List of interfaces
+        reverse (bool): Determines if list alphabetical list should be reversed. Defaults to False.
+
+    Returns:
+        list: Interfaces listed in alphabetical order.
+
+    Example:
+        >>> from netutils.interface import list_alphabetical
+        >>> list_alphabetical(["GigabitEthernet0/0", "GigabitEthernet0/1", "FastEthernet0/1", "FasEthernet0/0"])
+        ['FasEthernet0/0', 'FastEthernet0/1', 'GigabitEthernet0/0', 'GigabitEthernet0/1']
+        >>> list_alphabetical(["GigabitEthernet0/0", "GigabitEthernet0/1", "FastEthernet0/1", "FasEthernet0/0", "Ethernet1/0/1", "Ethernet0/0/1"])
+        ['Ethernet0/0/1', 'Ethernet1/0/1', 'FasEthernet0/0', 'FastEthernet0/1', 'GigabitEthernet0/0', 'GigabitEthernet0/1']
+        >>>
+    """
+    separate_interfaces_list = []
+    contained_interfaces = set()
+
+    for interface in interfaces:
+        interface_name = split_interface(interface)[0]
+        contained_interfaces.add(interface_name)
+
+    for interface_name in contained_interfaces:
+        new_list = [x for x in interfaces if split_interface(x)[0] == interface_name]
+        if new_list:
+            separate_interfaces_list.append(new_list)
```

Comment:
I sent you a starter snippet to simplify this a bit.  Leaving a note for others.  I'll poke at it some more in the morning.